File: 0345: reverseVowels.cpp
class Solution {
public:
    string reverseVowels(string s) {
        
        vector<int> indx;
        vector<char> vows;
        string check = "AEIOUaeiou";
        for(int i = 0; i < s.size(); i++){

            if(check.find(s[i]) != -1){
                indx.push_back(i);
                vows.push_back(s[i]);
            }
        }

        for(int i = 0; i<indx.size(); i++){

            int cur = indx.size()-1-i;
            s[indx[cur]] = vows[i];
        }

        return s;
    }
};

File: 0345: reverseVowels.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        vowelStack = []
        ans = ''
        vowels = "aeiouAEIOU"
        for x in s:
            if x in vowels:
                vowelStack.append(x)
        for x in s:
            if x in vowels:
                ans += vowelStack.pop()
            else:
                ans += x
        return ans